samcli/commands/build/command.py

```diff
@@ -54,73 +57,103 @@
 """
 
 
-@configuration_option(provider=TomlProvider(section="parameters"))
 @click.command("build", help=HELP_TEXT, short_help="Build your Lambda function code")
-@click.option('--build-dir', '-b',
-              default=DEFAULT_BUILD_DIR,
-              type=click.Path(file_okay=False, dir_okay=True, writable=True),  # Must be a directory
-              help="Path to a folder where the built artifacts will be stored. This directory will be first removed before starting a build.")
-@click.option("--base-dir", "-s",
-              default=None,
-              type=click.Path(dir_okay=True, file_okay=False),  # Must be a directory
-              help="Resolve relative paths to function's source code with respect to this folder. Use this if "
-                   "SAM template and your source code are not in same enclosing folder. By default, relative paths "
-                   "are resolved with respect to the SAM template's location")
-@click.option("--use-container", "-u",
-              is_flag=True,
-              help="If your functions depend on packages that have natively compiled dependencies, use this flag "
-                   "to build your function inside an AWS Lambda-like Docker container")
-@click.option("--manifest", "-m",
-              default=None,
-              type=click.Path(),
-              help="Path to a custom dependency manifest (ex: package.json) to use instead of the default one")
+@configuration_option(provider=TomlProvider(section="parameters"))
```

Comment:
+1 Lets add a note on via configuration_option needs to be present at the top of the decorator chain below click.command